tests/dragonfly/replication_test.py

```diff
@@ -226,3 +234,88 @@ async def disconnect(replica, c_replica, crash_type):
 
     # Check master survived all disconnects
     assert await c_master.ping()
+
+
+"""
+Test stopping master during different phases.
+
+This test is targeted at the replica cancellation mechanism that should quickly abort a failed operation
+and revert to connection retry state.
+
+Three types are tested:
+1. Master crashing during full sync state
+2. Master crashing in a random state.
+3. Master crashing during stable sync state
+
+"""
```

Comment:
This should reliably make sure that the replica can unblock and abort from almost any state